compiler-core/src/ast/typed.rs

```diff
@@ -76,6 +76,12 @@ pub enum TypedExpr {
         type_: Arc<Type>,
         fun: Box<Self>,
         arguments: Vec<CallArg<Self>>,
+        /// Whether the call has the right number and type of arguments.
+        /// This is useful for the language server to know if a call did type
+        /// check to offer some specific actions. Otherwise it would have to
+        /// perform type checking again.
+        ///
+        is_well_typed: bool,
```

Comment:
I'm not a huge fan of this. I think I would prefer it's a best-attempt that could be wrong when there's a type error, which is what we do for all the other code actions today.

Was there a reason you thought this code action would particularly want this extra check?